File: framework/wcf/Scenario/Federation/CS/Common/ISecurityTokenService.cs
//-----------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//-----------------------------------------------------------------------------
using System;
using System.ServiceModel;
using System.ServiceModel.Channels;

namespace Microsoft.Samples.Federation
{
	[ServiceContract(Name = "SecurityTokenService", Namespace = "http://tempuri.org")]
	public interface ISecurityTokenService
	{
        [OperationContract(Action = Constants.Trust.Actions.Issue,
                           ReplyAction = Constants.Trust.Actions.IssueReply)]        
		Message ProcessRequestSecurityToken(Message message);
	}
}

File: framework/wcf/Scenario/Federation/VB/Common/ISecurityTokenService.vb
' Copyright (c) Microsoft Corporation.  All Rights Reserved.

Imports System

Imports System.ServiceModel
Imports System.ServiceModel.Channels

Namespace Microsoft.Samples.Federation

    <ServiceContract(Name:="SecurityTokenService", [Namespace]:="http://tempuri.org")> _
    Public Interface ISecurityTokenService

        <OperationContract(Action:=Constants.Trust.Actions.Issue, ReplyAction:=Constants.Trust.Actions.IssueReply)> _
        Function ProcessRequestSecurityToken(ByVal message As Message) As Message

    End Interface

End Namespace

